internal/backends/winit/renderer/boxshadowcache.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright © SixtyFPS GmbH <info@slint-ui.com>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-commercial
+
+use std::{cell::RefCell, collections::BTreeMap};
+
+use i_slint_core::{items::ItemRc, Color};
+
+#[derive(Clone, PartialEq, Debug, Default)]
+pub struct BoxShadowOptions {
+    pub width: f32,
+    pub height: f32,
+    pub color: Color,
+    pub blur: f32,
+    pub radius: f32,
+}
+
+impl Eq for BoxShadowOptions {}
+impl Ord for BoxShadowOptions {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
```

Comment:
Why does this need a manual implementation?

Reply:
Because `f32` doesn't implement `Ord` by itself :(. I took this "trick" from the femtovg crate, where it's also used to put a struct with `f32` fields into a `BTreeMap`.